src/types/api.js

```diff
@@ -17,6 +17,32 @@
  * @prop {string} text
  * @prop {string} uri
  * @prop {string} user
+ * @prop {boolean} hidden
+ * @prop {boolean} [$highlight]
+ * @prop {boolean} [$orphan]
+ * @prop {boolean} [$anchorTimeout]
+ *
+ * @prop {Object} document
+ *   @prop {string} document.title
+ *
+ * @prop {Object} permissions
+ *   @prop {string[]} permissions.read
+ *   @prop {string[]} permissions.update
+ *   @prop {string[]} permissions.delete
+ *
+ * @prop {Object[]} target
+ *   @prop {string} target.source
+ *   @prop {Object[]} target.selector
```

Comment:
```suggestion
 *   @prop {Object[]} [target.selector]
```

The `selector` property is optional. Page notes do not have it.

Reply:
Sure. 
My method so far has been to make things not-optional until TS complains. I'm guessing as we fill things in more, these errors will popup and then we can go back and make things optional as they are needed.  Because you have knowledge of this, we can do it now, but if I'm to liberal here, and for example, I make a prop optional that really should not be, then we'll never get an error thats its incorrect unless we just visually catch it.